build/wix/mixxx.wxs

```diff
@@ -0,0 +1,429 @@
+ 
+  <!-- The URL for add/remove programs -->
+  <?define InfoURL="http://mixxx.org/" ?>
+   
+  <!-- 32-bit / 64-bit variables -->
+  <?if $(var.Platform) = x64 ?>
+    <?define Win64 = "yes" ?>
+    <?define arch = "64" ?>
+    <?define notarch = "32" ?>
+    <?define PlatformProgramFilesFolder = "ProgramFiles64Folder" ?>
+    <?define baseBuildDir = "..\.." ?>
+  <?else ?>
+    <?define Win64 = "no" ?>
+    <?define arch = "32" ?>
+    <?define notarch = "64" ?>
+    <?define PlatformProgramFilesFolder = "ProgramFilesFolder" ?>
+    <?define baseBuildDir = "..\.." ?>
+  <?endif ?>
+ 
+  <?define SourceDir="$(var.baseBuildDir)\dist$(var.arch)"?>
+  
+  <!--
+    ====================================================================================
+    Package start
+  -->
+
+  <!-- The upgrade code must never change as long as the product lives! -->
+  <!-- Product IDs must be autogenerated (*) or else major upgrades will not work -->
+  <Product Id="*"
+           Name="!(loc.ApplicationName)"
+           Language="!(loc.Language)"
+           Version="$(var.VersionNumber)"
+           Manufacturer="!(loc.ManufacturerFullName)"
+           UpgradeCode="$(var.UpgradeCode)" >
+   
+    <!-- Package IDs are valid for a single package version only - they are autogenerated by WiX -->
+    <!-- Let's require Windows Installer 4.0 (included in Vista) -->
+    <!-- And ALWAYS install per machine!!! -->
+    <Package Id="*"
+             Keywords="Installer"
+             InstallerVersion="400"
+             Compressed="yes"
+             InstallScope="perMachine"
+             Description="!(loc.ProductDescription)"
+             Manufacturer="!(loc.ManufacturerFullName)"
+             Comments="!(loc.Comments) $(var.VersionNumber)"
+             Platform="$(var.Platform)" />
+
+    <!-- Define icons (ID should not be longer than 18 chars and must end with ".exe") -->
+    <Icon Id="Icon.exe" SourceFile="$(var.baseBuildDir)\res\images\ic_mixxx.ico" />             
+    <Icon Id="Manual.ico" SourceFile="images\manual.ico" />             
+    
+    <!-- Set properties for add/remove programs -->
+    <Property Id="ARPPRODUCTICON" Value="Icon.exe" />
+    <Property Id="ARPHELPLINK" Value="$(var.InfoURL)" />
+    <Property Id="ARPNOREPAIR" Value="yes" Secure="yes" />      <!-- Remove repair -->
+    <Property Id="ARPNOMODIFY" Value="yes" Secure="yes" />      <!-- Remove modify -->
+
+    <!-- Upgrade logic -->
+    <!-- AllowSameVersionUpgrades -> Always upgrade, never allow two versions to be installed next to each other -->
+    <!-- AllowSameVersionUpgrades causes ICE61 which must be ignored -->
+    <MajorUpgrade DowngradeErrorMessage="!(loc.NewerInstalled)" AllowSameVersionUpgrades="yes" />
+
+    <Property Id="VCREDISTINSTALLED">
+      <RegistrySearch Id="VCRedistInstalledSearch"
+                      Root="HKLM"
+                      Key="SOFTWARE\Wow6432Node\Microsoft\VisualStudio\12.0\VC\Runtimes\$(var.Platform)"
```

Comment:
Right, I understand it's the VC version. I just want to get away from hard-coding that (as well as the 32-bit (Wow6432Node) assumption) in build scripts because it causes problems for anyone building Mixxx themselves on a system that doesn't have the exact same version installed. (And it causes problems for us when we upgrade the build environment.)
Even though NSIS does that now, does WiX offer a more graceful way to find this information?